content/articles/understanding-different-types-of-menus-in-android-with-kotlin/index.md

```diff
@@ -0,0 +1,261 @@
+                background.setBackgroundColor(Color.parseColor("#3c3f41"))
+                true
+            }
+            R.id.description -> {
+                var textView = findViewById<TextView>(R.id.textHappyCoding)
+                textView.text = "From Hello world to changing the world"
+                true
+            }
+            R.id.delete -> {
+                var textView = findViewById<TextView>(R.id.textHappyCoding)
+                textView.text = ""
+                true
+            }
+            else -> false
+        }
+    }
+```
+4. Checkable Menus 
+
+Checkable menus are frequently used in user interfaces to execute specific tasks, such as activating and deactivating features. For several executable actions, a check box or even a radio button can be used.
+
+checkable menu can be designed as follow in the resource file with the all required properties.
+```xml
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <group android:checkableBehavior="single">
+        <item
+            android:id="@+id/checkbox"
+            android:title="checkable_one" android:checked="false"/>
+        <item
+            android:id="@+id/checkbox2"
+            android:title="checkable_two"
+            android:checked="false"/>
+        <item
+            android:id="@+id/checkbox3"
+            android:title="checkable_three"
+            android:checked="false"/>
+    </group>
+    <group android:id="@+id/intent_based">
+
+    </group>
+
+</menu>
+```
+`checkablebehaviour` 
+
+is a property that can be used to indicate the check behavior and actions on checkable menu items. It can be specified in the following way.
+- `all`  
+This is used to demonstrate that all of the things may be selected, and a checkbox is used to facilitate this operation.
+- `single`
+This is used to show that just one item may be picked, and the radio button is used to do so.
+- `none`
+It is used to indicate no menu item is selected.
+
+just like options menu checkable menus click listeners are also handled within the `onOptionsItemSelected` method and it is done as shown below
+```kotlin
+override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        item.isChecked = !item.isChecked
+        return when (item.itemId) {
+            R.id.checkbox -> {
+                Toast.makeText(this, "checkable_one", Toast.LENGTH_SHORT).show()
+                true
+            }
+            R.id.checkbox2 -> {
+                Toast.makeText(this, "checkable_Two", Toast.LENGTH_SHORT).show()
+                true
+            }
+            R.id.checkbox3 -> {
+                Toast.makeText(this, "checkable_Three", Toast.LENGTH_SHORT).show()
+                true
+            }
+            else -> false
+        }
+    }
+```
+If a menu item is checked, it is first examined to see if it is checked, then unchecked so that the other may be checked using this code. `item.isChecked = !item.isChecked`
+
+5. Intent-Based menus
+
```

Comment:
kindly include at least one image for every type of menu. It's hard to know what to expect without a visual presentation.